cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionOption.cpp

```diff
@@ -0,0 +1,249 @@
+namespace ArcGISRuntime
+{
+namespace Toolkit
+{
+
+CoordinateConversionOption::CoordinateConversionOption(QObject* parent):
+  QObject(parent)
+{
+}
+
+/*!
+   \brief The destructor.
+ */
+CoordinateConversionOption::~CoordinateConversionOption()
+{
+}
+
+CoordinateConversionOption::CoordinateType CoordinateConversionOption::outputMode() const
+{
+  return m_outputMode;
+}
+
+void CoordinateConversionOption::setOutputMode(CoordinateType outputMode)
+{
+  if (outputMode == m_outputMode)
+    return;
+
+  m_outputMode = outputMode;
+  emit outputModeChanged();
+}
+
+QString CoordinateConversionOption::name() const
+{
+  return m_name;
+}
+
+void CoordinateConversionOption::setName(const QString& name)
+{
+  if (name == m_name)
+    return;
+
+  m_name = name;
+  emit nameChanged();
+}
+
+bool CoordinateConversionOption::hasSpaces() const
+{
+  return m_hasSpaces;
+}
+
+void CoordinateConversionOption::setHasSpaces(bool hasSpaces)
+{
+  if (m_hasSpaces == hasSpaces)
+    return;
+
+  m_hasSpaces = hasSpaces;
+  emit hasSpacesChanged();
+}
+
+int CoordinateConversionOption::precision() const
+{
+  return m_precision;
+}
+
+void CoordinateConversionOption::setPrecision(int precision)
+{
+  if (m_precision == precision)
+    return;
+
+  m_precision = precision;
+  emit precisionChanged();
+}
+
+int CoordinateConversionOption::decimalPlaces() const
+{
+  return m_decimalPlaces;
+}
+
+void CoordinateConversionOption::setDecimalPlaces(int decimalPlaces)
+{
+  if (m_decimalPlaces == decimalPlaces)
+    return;
+
+  m_decimalPlaces = decimalPlaces;
+  emit decimalPlacesChanged();
+}
+
+MgrsConversionMode CoordinateConversionOption::mgrsConversionMode() const
+{
+  return m_mgrsConversionMode;
+}
+
+void CoordinateConversionOption::setMgrsConversionMode(MgrsConversionMode mgrsConversionMode)
+{
+  if (m_mgrsConversionMode == mgrsConversionMode)
+    return;
+
+  m_mgrsConversionMode = mgrsConversionMode;
+  emit mgrsConversionModeChanged();
+}
+
+LatitudeLongitudeFormat CoordinateConversionOption::latLonFormat() const
+{
+  return m_latLonFormat;
+}
+
+void CoordinateConversionOption::setLatLonFormat(LatitudeLongitudeFormat latLonFormat)
+{
+  if (m_latLonFormat == latLonFormat)
+    return;
+
+  m_latLonFormat = latLonFormat;
+  emit latLonFormatChanged();
+}
+
+UtmConversionMode CoordinateConversionOption::utmConversionMode() const
+{
+  return m_utmConversionMode;
+}
+
+void CoordinateConversionOption::setUtmConversionMode(UtmConversionMode utmConversionMode)
+{
+  if (m_utmConversionMode == utmConversionMode)
+    return;
+
+  m_utmConversionMode = utmConversionMode;
+  emit utmConversionModeChanged();
+}
+
+GarsConversionMode CoordinateConversionOption::garsConversionMode() const
+{
+  return m_garsConvesrionMode;
+}
+
+void CoordinateConversionOption::setGarsConversionMode(GarsConversionMode conversionMode)
+{
+  if (m_garsConvesrionMode == conversionMode)
+    return;
+
+  m_garsConvesrionMode = conversionMode;
+  emit garsConversionModeChanged();
```

Comment:
I am on the fence here! I am not sure myself? Do you think it'll be worth it in the long term?

Reply:
Let's keep what you have for now. It will be easier to read and debug for us and users. If we need to write a lot more of these, we can consider a helper of some kind.